### PR TITLE
Fix Kubernetes cluster node count

### PR DESCRIPTION
**Pull request by xlucas**
### What does this PR do?
Fix the node count for Kubernetes clusters after the refactoring in https://github.com/DataDog/datadog-agent/pull/10274 dropped it.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nammn — APPROVED** [review]
Because we don't have a "transformer" we don't have any tests for the clusters resource. We should look into adding one in some other way (can be in this PR or in a follow-up)

File: releasenotes-dca/notes/bugfix-cluster-node-count-7d28105c0b7ff347.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG-DCA.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - Fix the node count reported for Kubernetes clusters
```
Comment (maycmlee) [inline releasenotes-dca/notes/bugfix-cluster-node-count-7d28105c0b7ff347.yaml:10]:
```suggestion
  - Fix the node count reported for Kubernetes clusters.
```

**Review by maycmlee — APPROVED** [review]
Small nit!

File: releasenotes-dca/notes/bugfix-cluster-node-count-7d28105c0b7ff347.yaml (hunk 1 above)
Comment (xlucas) [reply releasenotes-dca/notes/bugfix-cluster-node-count-7d28105c0b7ff347.yaml]: thanks!

**Pull request merged**